lab-dan/routes/dog-routes.js

```diff
@@ -0,0 +1,76 @@
+'use strict'
+
+const Dog = require('../model/dog')
+const Owner = require('../model/owner')
+const ObjectId = require('mongoose').Types.ObjectId
+
+module.exports = (router) => {
```

Comment:
Rather than relying on dependency injection here, you can pass your routes into `app.use` in the server.js file, and instantiate a router within each of these routes files. I 'think' that might fix your circular dependency issues (would have to test it out).